### ISSUE TITLE
Add `types.KindSAMLIdPServiceProvider` read only access to default implicit rule

### ISSUE THREAD

**Issue by flyinghermit** [gravitational/teleport#43379]
### What would you like Teleport to do?

Add `types.KindSAMLIdPServiceProvider` read only rule to default implicit role so users get read only access to SAML applications, similar to other resources such as Apps, Nodes etc. 

### What problem does this solve?

Better ux for SAML admins. 

### If a workaround exists, please include it.

Currently it requires explicit access to SAML resource:
```
allow:
    rules:
    - resources:
      - saml_idp_service_provider
      verbs:
      - list
      - read
```

**Comment by flyinghermit** [comment]
This is being worked and tracked in https://github.com/gravitational/teleport.e/issues/4723.

tl;dr: `types.KindSAMLIdPServiceProvide` will be supported in default implicit rule starting `v17`.
